### PR TITLE
osbuilder/scripts: fix yq 2.x failed to get rust version in yaml

### PR DESCRIPTION
**Pull request by Kvasscn**
yq 2.x version not support -X flag in read command, so remove it.

Fixes: #2824

Signed-off-by: zhanghj <zhanghj.lc@inspur.com>

### CODE REVIEW

File: tools/osbuilder/scripts/lib.sh
Comment (Jakob-Naucke) [reply tools/osbuilder/scripts/lib.sh:349]: I agree.

**Review by Jakob-Naucke — CHANGES_REQUESTED** [review]
@Kvasscn DYT this needs a backport? Also, as we are using this functionality, I think the answer is not to remove that flag, but to not allow <3 (nvm it would be best to move to v4 syntax in general).

**Comment by fidencio** [comment]
> @Kvasscn DYT this needs a backport? Also, as we are using this functionality, I think the answer is not to remove that flag, but to not allow <3 (nvm it would be best to move to v4 syntax in general).

Yep, we'd be better just moving to v4 syntax as soon as possible.

**Comment by jodh-intel** [comment]
Any update on this?

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by fidencio** [comment]
As this PR has been waiting for an update since October 2021, I'm closing it for now.

@Kvasscn, please, if you find some free cycles to work on this PR, please, just re-open it and let us know.
Thanks for the contribution and looking forward to seeing an up-to-date version of this PR.

**Pull request closed without merging**
